File: apps/api/app/services/search_service.py
from app.repositories.mock_search_repository import MockSearchRepository
from app.schemas.search import SearchSuggestion


class SearchService:
    def __init__(self, repository: MockSearchRepository | None = None) -> None:
        self.repository = repository or MockSearchRepository()

    def search_stocks(self, query: str) -> list[SearchSuggestion]:
        return self.repository.search_stocks(query)

File: apps/web/services/search-service.ts
import { MockSearchRepository } from "@/repositories/mock/mock-search-repository";
import type { SearchSuggestion } from "@/types/search";

const searchRepository = new MockSearchRepository();

export const searchService = {
  searchStocks(query: string): Promise<SearchSuggestion[]> {
    return searchRepository.searchStocks(query);
  }
};

